process/track/shardBlockTrack.go

```diff
@@ -92,8 +84,8 @@ func NewShardBlockTrack(arguments ArgShardTracker) (*shardBlockTrack, error) {
 	sbt.blockProcessor = blockProcessor
 
 	sbt.headers = make(map[uint32]map[uint64][]*headerInfo)
-	sbt.shardHeadersPool.RegisterHandler(sbt.receivedShardHeader)
-	sbt.metaBlocksPool.RegisterHandler(sbt.receivedMetaBlock)
+	sbt.headersPool.RegisterHandler(sbt.receivedShardHeader)
```

Comment:
Now, because both methods are notified when a new shards/meta header is received we should put a filter in each method (in receivedMetaBlock to avoid headers with shardId <> MetaChain and in receivedShardHeader to avoid headers with shardId == MetaChain)

Reply:
done.